### ISSUE TITLE
[Feature Request] Own Context Menu implementation with runtime editing

### ISSUE THREAD

**Issue by Bluscream** [pathmann/pyTSon#11]
I know you had this on your radar already but i want to have it written down somewhere so you could assign a milestone.

Some people would love to have the context menus reimplemented in a different way then the TS API provides.

So you can change values at runtime. (To assign groups for example)
And to be able to deactivate (grey out) items that are not available (For your own client for example).

Thanks in advance :)

**Comment by pathmann** [comment]
The client's main menu (global plugin menuitems) are no problem.

The problem starts with the TreeView's context menu. This needs a little testing. If the underlying QMenu is dynamically created on each click, it will become messy.
If the menu is created only once, it should be possible.

**Comment by Bluscream** [comment]
Also wished:

- Horizontal spacer lines
- Sub menus (per script if it has more than 1 entry)
- adding entries (almost) everywhere

**Comment by pathmann** [comment]
Short answer: Not possible, if you need this, request it for the plugin sdk and it can be then ported into pyTSon.

Long answer: I looked into it and decided to not change it. As I already mentioned, the global menu is no problem. But the client and channel context menus are created on demand (or at least not as child of any accessible QObject-based class). So to access them, you need to "wait" for them to pop up and then manipulate them, so that your items are available.
Anonther solution would be to completely replace the client's context menus. But that would be even worse.
Any pyTSon plugin developer can implement solution 1 and/or 2 (or any other) in his script, but this won't happen in pyTSon's core.

**Comment by Bluscream** [comment]
> Any pyTSon plugin developer can implement solution 1 and/or 2 (or any other) in his script, but this won't happen in pyTSon's core.

You as c++ dev say you won't do it but expect scripters to do it? lol :D anyway, sure it would have been to good to be real ^^
